.circleci/config.yml

```diff
@@ -1,9 +1,26 @@
-version: 2.0
-jobs:
-  build:
+version: 2.1
+executors:
+  docker-executor:
     docker:
       - image: outpostuniverse/ubuntu-18.04-gcc-sdl2-physfs-circleci
+  macos-executor:
+    macos:
+      xcode: "10.0.0"
```

Comment:
I looked quickly into what `executors` are. I'm not sure we're actually really using this feature in any special way. If the `docker`/`macos` sections were moved below the specific jobs, I think it would function identically.

This leaves me feeling like I've missed something about how to effectively use `executors`.

Reply:
It's also useful for organization and readability. By having the environment definition separated out from the build job it reduces noise and makes it easier to read and follow what the job is supposed to be doing.

Reply:
From what I read, executors are useful when you re-use an environment. Here we are only using each environment once, so we're not really benefiting from that.

I'm not sure it does anything for readability here.

I think readability might be better improved by pulling out the build and test commands and re-using that. That's where the duplication really is.

Reply:
Adding test jobs now makes using executors "useful".

Reply:
Huh, I wouldn't have thought to use executors like that. I would have just appended the test step after the build step.

I see using multiple executors results in more result output in the GitHub interface. It now reports build and test pass/fail results separately.

Actually, I think it spins up a new Docker instance for the tests when you do this. That likely means it would need to compile the code again in the new instance before running tests. It also doubles the container startup time.

We may need to do this for branch versus PR builds, but I don't think we should be doing this for build and test. Build and test should both be done in the same container.